src/provider/states.rs

```diff
@@ -22,3 +26,47 @@ macro_rules! fail_fatal {
         return Transition::Complete(Err(aerr));
     }};
 }
+
+/// Create basic Pod status patch with container status and pod conditions
+pub fn make_status_with_containers_and_condition(
+    phase: Phase,
+    reason: &str,
+    container_statuses: Vec<KubeContainerStatus>,
+    init_container_statuses: Vec<KubeContainerStatus>,
+    pod_conditions: Vec<KubePodCondition>,
+) -> serde_json::Value {
+    serde_json::json!(
+       {
+           "metadata": {
+               "resourceVersion": "",
```

Comment:
Why the empty field?

Reply:
I simply copied this from the Krustlet code. Just done a quick test and it work fine without the empty field, as far as I can tell.